servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -152,30 +151,25 @@ final class DefaultLoadBalancer<ResolvedAddress, C extends LoadBalancedConnectio
             final LoadBalancerObserverFactory loadBalancerObserverFactory,
             @Nullable final HealthCheckConfig healthCheckConfig,
             final Function<String, OutlierDetector<ResolvedAddress, C>> outlierDetectorFactory) {
-        this.lbDescription = makeDescription(
-                requireNonNull(id, "id"), requireNonNull(targetResource, "targetResource"));
-        this.subsetter = subsetterFactory.newSubsetter(lbDescription);
-        this.hostSelector = requireNonNull(loadBalancingPolicy, "loadBalancingPolicy")
-                .buildSelector(Collections.emptyList(), lbDescription);
-        this.priorityStrategy = requireNonNull(
-                priorityStrategyFactory, "priorityStrategyFactory").apply(lbDescription);
-        this.connectionSelector = requireNonNull(connectionSelectorPolicy,
-                "connectionSelectorPolicy").buildConnectionSelector(lbDescription);
-        this.eventPublisher = requireNonNull(eventPublisher);
+        this.lbDescription = makeDescription(id, targetResource);
+        this.subsetter = requireNonNull(subsetterFactory.newSubsetter(lbDescription));
+        this.hostSelector = requireNonNull(loadBalancingPolicy.buildSelector(emptyList(), lbDescription));
+        this.priorityStrategy = requireNonNull(priorityStrategyFactory.apply(lbDescription));
+        this.connectionSelector = requireNonNull(connectionSelectorPolicy.buildConnectionSelector(lbDescription));
+        this.eventPublisher = eventPublisher;
         this.eventStream = fromSource(eventStreamProcessor)
                 .replay(1); // Allow for multiple subscribers and provide new subscribers with last signal.
-        this.connectionFactory = requireNonNull(connectionFactory);
+        this.connectionFactory = connectionFactory;
         this.minConnectionsPerHost = minConnectionsPerHost;
         this.loadBalancerObserver = CatchAllLoadBalancerObserver.wrap(
-                requireNonNull(loadBalancerObserverFactory, "loadBalancerObserverFactory")
-                .newObserver(lbDescription));
+                loadBalancerObserverFactory.newObserver(lbDescription));
         this.healthCheckConfig = healthCheckConfig;
         this.sequentialExecutor = new SequentialExecutor((uncaughtException) ->
                 LOGGER.error("{}: Uncaught exception in {}", this, this.getClass().getSimpleName(), uncaughtException));
         this.asyncCloseable = toAsyncCloseable(this::doClose);
         // Maintain a Subscriber so signals are always delivered to replay and new Subscribers get the latest signal.
         eventStream.ignoreElements().subscribe();
-        this.outlierDetector = requireNonNull(outlierDetectorFactory, "outlierDetectorFactory").apply(lbDescription);
+        this.outlierDetector = requireNonNull(outlierDetectorFactory.apply(lbDescription));
```

Comment:
Consider running all validation and using factories before starting any actual work, like `eventStream.ignoreElements().subscribe()`